packages/debian/download_packages.py

```diff
@@ -78,7 +78,9 @@ def fetch_binary(
     filename = pathlib.Path(package.filename).name
     destfile = destdir/filename
     print('Downloading package {}'.format(filename))
-    with urllib.request.urlopen(package.uri) as response:
+    pkg_uri = urllib.parse.urlparse(package.uri)
+    pkg_uri = pkg_uri._replace(path=urllib.parse.quote(pkg_uri.path))
+    with urllib.request.urlopen(urllib.parse.urlunparse(pkg_uri)) as response:
```

Comment:
Why not simply:

```python
uri = urllib.parse.quote(package.uri)
with urllib.request.urlopen(uri) as response:
```
?

Also, if the problem is a bad char in the URL, why does it breaks just now?

URL changed from upstream recently? (if so, could be mentioned in the commit message, even giving an example of problematic URI would be cool)

Reply:
The current URL is:
```
http://repo.saltstack.com/apt/ubuntu/18.04/amd64/archive/2018.3.4/pool/main/s/salt/salt-minion_2018.3.4+ds-1_all.deb
```
Problem is the `+` character is interpreted as a space, so we need to escape it.

Can't quote the whole URI, otherwise it also escapes `:` after the scheme:
```
http%3A//repo.saltstack.com/apt/ubuntu/18.04/amd64/archive/2018.3.4/pool/main/s/salt/salt-minion_2018.3.4%2Bds-1_all.deb
```

I don't know if the URI changed recently, it seems not:
```
2019-10-03T22:34:59.000Z        26.9 kB        salt-minion_2018.3.4+ds-1_all.deb
```
Maybe the change comes from elsewhere, urllib, the remote server...  I'll investigate, I just wanted to fix it quickly first, as it blocks the builds.

Reply:
Ok, works for me.
But would be good to find where the change came from, to be sure this fix is the right one.

Reply:
It seems that's not related to `urllib`, as it succeed to fetch other packages with `+` sign in the URL.
I'm 99.9% sure there is a change on repo.saltstack.com on how it handles URL.

Reply:
Ok, easy to reproduce with a browser go to:
```
http://archive.ubuntu.com/ubuntu/pool/main/libz/libzstd/libzstd1_1.3.3+dfsg-2ubuntu1.1_amd64.deb
```
The package is downloaded.
Then, go to:
```
http://repo.saltstack.com/apt/ubuntu/18.04/amd64/archive/2018.3.4/pool/main/s/salt/salt-minion_2018.3.4+ds-1_all.deb
```
It does not work.

Reply:
I'll change the commit message for something more accurate.